saleor/cart/models.py

```diff
@@ -160,10 +174,10 @@ def create_line(self, variant, quantity, data):
         return line
 
     def get_line(self, variant, data=None):
-        try:
-            return self.lines.get(variant=variant)
-        except CartLine.DoesNotExist:
-            return None
+        all_lines = self.lines.all()
+        line = [line for line in all_lines if line.variant_id == variant.id]
```

Comment:
This is wrong but the previous implementation was also wrong. Neither takes `data` into account. Please make it only match on matching data and add a test that covers this.